modules/memstore_redis/variables.tf

```diff
@@ -49,11 +48,11 @@ variable "memory_size" {
 
 variable "replicas" {
   type        = number
-  default     = 1
-  description = "The number of instances to create"
+  default     = 0
+  description = "The number of read replicas to create"
   validation {
-    condition     = var.replicas >= 1 && var.replicas <= 5
-    error_message = "The valid range for the Standard Tier with read replicas enabled is [1-5] and defaults to 1."
+    condition     = var.replicas <= 5
+    error_message = "The valid range for the Standard Tier with read replicas enabled is [1-5] and defaults to 0 as the default is zero read-replicas."
```

Comment:
Since 1.9 you can do cross input variable references in input validations.  So can make conditional based on the tier here.

Reply:
Oh nice, that's nice to know. I rewrote the validations as preconditions on the resource.